src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/ParamServer.java

```diff
@@ -180,52 +197,65 @@ private void setFinishedState(int workerID) {
 			_finishedStates[workerID] = true;
 		}
 
+		private void broadcastModel() {
+			IntStream.range(0, _finishedStates.length).forEach(i -> _modelMap.compute(i, (id, q) -> {
+				if (q == null) {
+					q = new ArrayBlockingQueue<>(1);
+				}
+				try {
+					q.put(_model);
+				} catch (InterruptedException e) {
+					throw new DMLRuntimeException(
+							String.format("Param server: failed to broadcast the model for worker_%d", id), e);
+				}
+				return q;
+			}));
+		}
+
 		@Override
 		public void run() {
-			synchronized (_lock) {
-				while (isAlive()) {
```

Comment:
Please remove unused methods such as `isAlive()`; in case you need it later, we would need to make the alive flag volatile to ensure that updates are seen from all threads (unless used in a synchronized which implicitly creates a memory barrier as well).